inc/infocom.class.php

```diff
@@ -1955,11 +1956,20 @@ static function getWarrantyExpir($from, $addwarranty, $deletenotice = 0, $color
          return "";
       }
 
-      $datetime = strtotime("$from+$addwarranty month -$deletenotice month");
-      if ($color && ($datetime < time())) {
-         return "<span class='red'>".Html::convDate(date("Y-m-d", $datetime))."</span>";
+      $timestamp = strtotime("$from+$addwarranty month -$deletenotice month");
+
+      if ($auto_renew) {
```

Comment:
To prevent any infinite loop on bad usage of this method.
```suggestion
      if ($auto_renew && $addwarranty > 0) {
```